pkg/app/piped/executor/ecs/ecs.go

```diff
@@ -174,6 +186,31 @@ func applyServiceDefinition(ctx context.Context, cli provider.Client, serviceDef
 	return service, nil
 }
 
+func findRemovedTags(currentTags, desiredTags []types.Tag) []string {
+	var tagsToRemove []string
+
+	// Avoid removing PipeCD-managed tags, even though they're usually set in loadServiceDefinition()
+	for _, t := range currentTags {
+		if *t.Key == provider.LabelManagedBy || *t.Key == provider.LabelPiped || *t.Key == provider.LabelApplication || *t.Key == provider.LabelCommitHash {
```

Comment:
[nit] This comment is related to only the if-clause

```suggestion
	for _, t := range currentTags {
		// Avoid removing PipeCD-managed tags, even though they're usually set in loadServiceDefinition()
		if *t.Key == provider.LabelManagedBy || *t.Key == provider.LabelPiped || *t.Key == provider.LabelApplication || *t.Key == provider.LabelCommitHash {
```